soco/exceptions.py

```diff
@@ -80,3 +80,37 @@ class SoCoSlaveException(SoCoException):
 
 class NotSupportedException(SoCoException):
     """Raised when something is not supported by the device"""
+
+
+class SoCoFault(object):
+    """Class to represent a failed object instantiation.
+
+    It rethrows the exception on common use.
+    """
+
+    def __init__(self, exception, **kwargs):
+        """
+        Args:
+            exception (Exception): The exception which should be thrown on use
+            kwargs: Items to add to the object dict, as additional information
+                about the fault.
+        """
+        self.__dict__ = kwargs
+        self.__dict__['exception'] = exception
+
+    def __getattr__(self, name):
+        raise self.exception
+
+    def __setattr__(self, name, value):
+        raise self.exception
+
+    def __getitem__(self, item):
+        raise self.exception
+
+    def __repr__(self):
+        return '<{0}: {1} at {2}>'.format(self.__class__.__name__,
+                                          repr(self.exception),
+                                          hex(id(self)))
+
+    def __str__(self):
```

Comment:
When inheriting from object, __str__ seems to default to calling __repr__, so this methods seems unnecessary.

Reply:
That was actually a stub, it has a different implementation now.